features/dd-sdk-android-rum/src/main/kotlin/com/datadog/android/rum/internal/instrumentation/gestures/GesturesListener.kt

```diff
@@ -34,19 +37,24 @@ internal class GesturesListener(
     private val contextRef: Reference<Context>,
     private val internalLogger: InternalLogger,
     private val composeActionTrackingStrategy: ActionTrackingStrategy = NoOpActionTrackingStrategy(),
-    private val androidActionTrackingStrategy: ActionTrackingStrategy = AndroidActionTrackingStrategy(
-        internalLogger
-    )
+    private val androidActionTrackingStrategy: ActionTrackingStrategy = AndroidActionTrackingStrategy()
 ) : GestureListenerCompat() {
 
     private var scrollEventType: RumActionType? = null
     private var gestureDirection = ""
-    private var scrollTargetReference: WeakReference<View?> = WeakReference(null)
+    private var scrollTargetReference: WeakReference<ViewTarget?> = WeakReference(null)
     private var onTouchDownXPos = 0f
     private var onTouchDownYPos = 0f
 
     // region GesturesListener
 
+    init {
+        contextRef.get()?.let {
+            composeActionTrackingStrategy.register(sdkCore, it)
+            androidActionTrackingStrategy.register(sdkCore, it)
```

Comment:
normally we need to unregister as well, but I guess if SDK core is stopped, then it won't be referenced from user code and even though it will have a reference from action tracking strategies, it will be still deallocated.